test/bigtable/mutation.js

```diff
@@ -0,0 +1,387 @@
+      var message = true;
+      var notEncoded = Mutation.convertToBytes(message);
+
+      assert(!(notEncoded instanceof Buffer));
+      assert.strictEqual(message, notEncoded);
+    });
+  });
+
+  describe('createTimeRange', function() {
+    it('should create a time range', function() {
+      var timestamp = Date.now();
+      var dateObj = new Date(timestamp);
+      var range = Mutation.createTimeRange(dateObj, dateObj);
+
+      assert.strictEqual(range.startTimestampMicros, timestamp);
+      assert.strictEqual(range.endTimestampMicros, timestamp);
+    });
+  });
+
+  describe('encodeSetCell', function() {
+    var convert;
+    var convertCalls = [];
+
+    before(function() {
+      convert = Mutation.convertToBytes;
+      Mutation.convertToBytes = function(value) {
+        convertCalls.push(value);
+        return value;
+      };
+    });
+
+    after(function() {
+      Mutation.convertToBytes = convert;
+    });
+
+    beforeEach(function() {
+      convertCalls = [];
+    });
+
+    it('should encode a setCell mutation', function() {
+      var fakeMutation = {
+        follows: {
+          gwashington: 1,
+          alincoln: 1
+        }
+      };
+
+      var cells = Mutation.encodeSetCell(fakeMutation);
+
+      assert.strictEqual(cells.length, 2);
+
+      assert.deepEqual(cells, [{
+        setCell: {
+          familyName: 'follows',
+          columnQualifier: 'gwashington',
+          timestampMicros: -1,
+          value: 1
+        }
+      }, {
+        setCell: {
+          familyName: 'follows',
+          columnQualifier: 'alincoln',
+          timestampMicros: -1,
+          value: 1
+        }
+      }]);
+
+      assert.strictEqual(convertCalls.length, 4);
+      assert.deepEqual(convertCalls, ['gwashington', 1, 'alincoln', 1]);
+    });
+
+    it('should optionally accept a timestamp', function() {
+      var timestamp = Date.now();
+      var fakeMutation = {
+        follows: {
+          gwashington: {
+            value: 1,
+            timestamp: new Date(timestamp)
+          }
+        }
+      };
+
+      var cells = Mutation.encodeSetCell(fakeMutation);
+
+      assert.deepEqual(cells, [{
+        setCell: {
+          familyName: 'follows',
+          columnQualifier: 'gwashington',
+          timestampMicros: timestamp,
+          value: 1
+        }
+      }]);
+
+      assert.strictEqual(convertCalls.length, 2);
+      assert.deepEqual(convertCalls, ['gwashington', 1]);
+    });
+  });
+
+  describe('encodeDelete', function() {
+    var convert;
+    var convertCalls = [];
+
+    before(function() {
+      convert = Mutation.convertToBytes;
+      Mutation.convertToBytes = function(value) {
+        convertCalls.push(value);
+        return value;
+      };
+    });
+
+    after(function() {
+      Mutation.convertToBytes = convert;
+    });
+
+    beforeEach(function() {
+      convertCalls = [];
+    });
+
+    it('should create a delete row mutation', function() {
```

Comment:
I think it's fine to test it by just giving it a single item not in an array and doing a basic check to confirm it works. The important part is singling that out in a test so that if something changes later, the test that says "it should arrify the input" fails.